mono/utils/memfuncs.c

```diff
@@ -288,7 +288,22 @@ mono_determine_physical_ram_size (void)
 		return 134217728;
 	}
 
-	return page_size * num_pages;
+	memsize = page_size * num_pages;	/* Calculate physical memory size */
+
+	gint64 restricted_limit = getRestrictedPhysicalMemoryLimit();	/* Check for any cgroup limit */
+	if (restricted_limit != 0) {
+		gchar *heapHardLimit = getenv("DOTNET_GCHeapHardLimit");	/* See if user has set a limit */
+		if (heapHardLimit != NULL) {
+			guint64 gcLimit = strtoull(heapHardLimit, NULL, 16);
+			if (gcLimit != 0)					
+				restricted_limit = (restricted_limit < gcLimit ? restricted_limit : (gint64) gcLimit);
+		} else
+			restricted_limit = (3 * restricted_limit) / 4;	/* Use 75% limit of container */
```

Comment:
Can you read `DOTNET_GCHeapHardLimitPercent`, and use it as a way to configure the `3/4` ratio?

Reply:
The doc says: `DOTNET_GCHeapHardLimitPercent | hexadecimal value`. Is it an integer or can it contain a fractional part? I assume I should ignore if < 0 or > 100. I tried looking in the runtime source code but couldn't see it being used: i.e. `find . -type f -exec grep -il DOTNET_GCHeapHardLimitPercent {} \;` yields nothing.

Reply:
The config is defined here:
https://github.com/dotnet/runtime/blob/9a9a4f34ff79167a6b7649b85d9808a10fbba10a/src/coreclr/gc/gcconfig.h#L104

It is accessed via GCConfig::GetGCHeapHardLimitPercent().